src/masternodes/mn_rpc.cpp

```diff
@@ -1558,32 +2005,675 @@ UniValue accounttoutxos(const JSONRPCRequest& request) {
+                                                                            "\\\"tokenTo\\\":\\\"Token2\\\","
+                                                                            "\\\"maxPrice\\\":\\\"0.01\\\""
+                                                                            "}\" \"[{\\\"txid\\\":\\\"id\\\",\\\"vout\\\":0}]\" ")
+                             },
+              }.Check(request);
+
+
+    if (pwallet->chain().isInitialBlockDownload()) {
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot create transactions while still in Initial Block Download");
+    }
+
+    RPCTypeCheck(request.params, {UniValue::VOBJ, UniValue::VARR}, true);
+
+    CPoolSwapMessage poolSwapMsg{};
+    std::string tokenFrom, tokenTo;
+    UniValue metadataObj = request.params[0].get_obj();
+    if (!metadataObj["from"].isNull()) {
+        poolSwapMsg.from = DecodeScript(metadataObj["from"].getValStr());
+    }
+    if (!metadataObj["tokenFrom"].isNull()) {
+        tokenFrom = metadataObj["tokenFrom"].getValStr();
+    }
+    if (!metadataObj["amountFrom"].isNull()) {
+        poolSwapMsg.amountFrom = AmountFromValue(metadataObj["amountFrom"]);
+    }
+    if (!metadataObj["to"].isNull()) {
+        poolSwapMsg.to = DecodeScript(metadataObj["to"].getValStr());
+    }
+    if (!metadataObj["tokenTo"].isNull()) {
+        tokenTo = metadataObj["tokenTo"].getValStr();
+    }
+    {
+        LOCK(cs_main);
+        auto token = pcustomcsview->GetTokenGuessId(tokenFrom, poolSwapMsg.idTokenFrom);
+        if (!token)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "TokenFrom was not found");
+
+        auto token2 = pcustomcsview->GetTokenGuessId(tokenTo, poolSwapMsg.idTokenTo);
+        if (!token2)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "TokenTo was not found");
+
+        if (!metadataObj["maxPrice"].isNull()) {
+            CAmount maxPrice = AmountFromValue(metadataObj["maxPrice"]);
+            poolSwapMsg.maxPrice.integer = maxPrice / COIN;
+            poolSwapMsg.maxPrice.fraction = maxPrice % COIN;
+        }
+        else {
+            // This is only for maxPrice calculation
+
+            auto poolPair = pcustomcsview->GetPoolPair(poolSwapMsg.idTokenFrom, poolSwapMsg.idTokenTo);
+            if (!poolPair) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Can't find the poolpair " + tokenFrom + "-" + tokenTo);
+            }
+            CPoolPair const & pool = poolPair->second;
+            if (pool.totalLiquidity <= CPoolPair::MINIMUM_LIQUIDITY) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Pool is empty!");
+            }
+
+            auto resA = pool.reserveA;
+            auto resB = pool.reserveB;
+            if (poolSwapMsg.idTokenFrom != pool.idTokenA) {
+                std::swap(resA, resB);
+            }
+            arith_uint256 price256 = arith_uint256(resB) * CPoolPair::PRECISION / arith_uint256(resA);
```

Comment:
Before divide, need to check if resA is 0

Reply:
They are can't be zero "by nature" if "pool.totalLiquidity > CPoolPair::MINIMUM_LIQUIDITY", even after trading